app/src/main/java/com/example/countryweather/net/countries/CountryEndpointInterface.kt

```diff
@@ -0,0 +1,9 @@
+package com.example.countryweather.net.countries
+
+import retrofit2.Call
+import retrofit2.http.GET
+
+interface CountryEndpointInterface {
+    @GET("v2/all")
+    fun getCountriesList():Call<List<CountryDataItem>>
```

Comment:
Integrate RxJava with Retrofit and use `Single<List<CountryDataItem>>`

https://github.com/square/retrofit/tree/master/retrofit-adapters/rxjava2